Sources/StreamChat/Utils/Timers.swift

```diff
@@ -97,24 +98,30 @@ private class RepeatingTimer: RepeatingTimerControl {
         timer.cancel()
         // If the timer is suspended, calling cancel without resuming
         // triggers a crash. This is documented here https://forums.developer.apple.com/thread/15902
-        resume()
+        if state == .suspended {
```

Comment:
Didn't know you have to resume an already cancelled timer, that's counterintuitive 😅

Reply:
Yup, really weird 🤷‍♂️ This a really old API, we probably should move to something more modern, but for now it should work. If it doesn't we can reconsider it.